pkg/controller/networkpolicy/controller_querier_replier.go

```diff
@@ -20,119 +20,122 @@ package networkpolicy
 }
 
 // EndpointQueryReplier is responsible for handling query requests from antctl query
 type EndpointQueryReplier struct {
-	// addressGroupStore is the storage where the populated Address Groups are stored.
-	addressGroupStore storage.Interface
-	// appliedToGroupStore is the storage where the populated AppliedTo Groups are stored.
-	appliedToGroupStore storage.Interface
-	// internalNetworkPolicyStore is the storage where the populated internal Network Policy are stored.
-	internalNetworkPolicyStore storage.Interface
-	// podInformer is used to check existence of a selected pod
-	podInformer coreinformers.PodInformer
+	networkPolicyController *NetworkPolicyController
 }
 
 // EndpointQueryResponse is the reply struct for QueryNetworkPolicies
 type EndpointQueryResponse struct {
-	Endpoints []Endpoint
-	Error error
+	Endpoints []Endpoint `json:"endpoints,omitempty"`
 }
 
 // Endpoint holds response information for an endpoint following a query
 type Endpoint struct {
-	Namespace string
-	Name string
-	Policies []Policy
-	Rules []Rule
+	Namespace string   `json:"namespace,omitempty"`
+	Name      string   `json:"name,omitempty"`
+	Policies  []Policy `json:"policies,omitempty"`
+	Rules     []Rule   `json:"rules,omitempty"`
 }
 
 type PolicyRef struct {
-	Namespace string
-	Name string
-	UID types.UID
+	Namespace string    `json:"namespace,omitempty"`
+	Name      string    `json:"name,omitempty"`
+	UID       types.UID `json:"uid,omitempty"`
 }
 
 // Policy holds network policy information to be relayed to client following query endpoint
 type Policy struct {
 	PolicyRef
-	selector metav1.LabelSelector
+	selector metav1.LabelSelector `json:"selector,omitempty"`
 }
 
 // Rule holds
 type Rule struct {
 	PolicyRef
-	Direction networkingv1beta1.Direction
-	RuleIndex int
-	Ports []v1.NetworkPolicyPort
+	Direction networkingv1beta1.Direction `json:"direction,omitempty"`
+	RuleIndex int                         `json:"ruleindex,omitempty"`
+	Ports     []v1.NetworkPolicyPort      `json:"ports,omitempty"`
 }
 
 // NewNetworkPolicyController returns a new *NetworkPolicyController.
-func NewEndpointQueryReplier(
-	addressGroupStore storage.Interface,
-	appliedToGroupStore storage.Interface,
-	internalNetworkPolicyStore storage.Interface,
-	podInformer coreinformers.PodInformer) *EndpointQueryReplier {
+func NewEndpointQueryReplier(networkPolicyController *NetworkPolicyController) *EndpointQueryReplier {
 	n := &EndpointQueryReplier{
-		addressGroupStore:          addressGroupStore,
-		appliedToGroupStore:        appliedToGroupStore,
-		internalNetworkPolicyStore: internalNetworkPolicyStore,
-		podInformer: podInformer,
+		networkPolicyController: networkPolicyController,
 	}
 	return n
 }
 
 //Query functions
-func (eq EndpointQueryReplier) QueryNetworkPolicies(namespace string, podName string) *EndpointQueryResponse {
+func (eq EndpointQueryReplier) QueryNetworkPolicies(namespace string, podName string) (*EndpointQueryResponse, error) {
```

Comment:
do we have a test for the case where an endpoint isn't selected by any policy / rule? would be good to have (in this case I don't expect an error of course)

Reply:
I'll add one